tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -221,11 +235,22 @@ private[scalanative] object LLVM {
         if (config.targetsWindows) Seq("dbghelp")
         else if (config.targetsOpenBSD || config.targetsNetBSD)
           Seq("pthread")
-        else Seq("pthread", "dl")
+        else Seq("pthread", "dl", "m")
```

Comment:
This is an artifact of linking with `clang` instead of `clang++` on linux, probably `libm` was a dependency of `libc++` and we were importing it by accident.